src/utils/HealthLogger.js

```diff
@@ -0,0 +1,167 @@
+        PerfUtils                   = require("utils/PerfUtils"),
+
+        HEALTH_DATA_STATE_KEY       = "HealthData.Logs",
+        logHealthData               = true;
+
+    /**
+     * Init: creates the health log preference keys in the state.json file
+     */
+    function init() {
+        PreferencesManager.stateManager.definePreference(HEALTH_DATA_STATE_KEY, "object", {});
+    }
+
+    /**
+     * Return the Performance related data
+     * @returns {Object} Performance Data aggregated till now
+     */
+    function getPerformanceData() {
+        var perfData = PerfUtils.getHealthReport();
+        return perfData;
+    }
+
+    /**
+     * Return the aggregate of all health data logged till now from all sources
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getAggregatedHealthData() {
+        var healthData = PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+        jQuery.extend(healthData, getPerformanceData());
+        return healthData;
+    }
+
+    /**
+     * Return all health data logged till now stored in the state prefs
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getHealthData() {
+        var healthData = PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+        return healthData;
+    }
+
+    /**
+     * Sets the health data
+     * @param {Object} dataObject The object to be stored as health data
+     */
+    function setHealthData(dataObject) {
+        PreferencesManager.setViewState(HEALTH_DATA_STATE_KEY, dataObject);
+    }
+
+    /**
+     * Returns health data logged for the given key
+     * @returns {Object} Health Data object for the key or undefined if no health data stored
+     */
+    function getHealthDataLog(key) {
+        var healthData = getHealthData();
+        return healthData[key];
+    }
+
+    /**
+     * Sets the health data for the given key
+     * @param {Object} dataObject The object to be stored as health data for the key
+     */
+    function setHealthDataLog(key, dataObject) {
+        var healthData = getHealthData();
+        healthData[key] = dataObject;
+        setHealthData(healthData);
+    }
+
+    /**
+     * Clears all the health data recorded till now
+     */
+    function clearHealthData() {
+        PreferencesManager.setViewState(HEALTH_DATA_STATE_KEY, {});
+        //clear the preformance relaed health data also
+        PerfUtils.clear();
+    }
+
+    /**
+     * Enable or disable health data logs
+     * @param {Boolean} enabled true to enable health logs
+     */
+    function setHealthLogsEnabled(enabled) {
+        logHealthData = enabled;
+        if (!enabled) {
+            clearHealthData();
+        }
+    }
+
+    /**
+     * All the logging functions should be disabled if this returns false
+     * @returns {Boolean} true if health data can be logged
+     */
+    function shouldLogHealthData() {
+        return logHealthData;
+    }
```

Comment:
This function is a simple getter that is used only once and also not being exported, so we could eliminate it and use `logHealthData` where `shouldLogHealthData` is being called.